presto-spi/src/main/java/com/facebook/presto/spi/ColumnMetadata.java

```diff
@@ -91,6 +90,20 @@ public Map<String, Object> getProperties()
         return properties;
     }
 
+    public ColumnMetadata copy(java.util.function.Function<Builder, Builder> customizer)
+    {
+        Builder builder = ColumnMetadata.builder()
+                .setName(getName())
+                .setType(getType())
+                .setNullable(isNullable())
+                .setComment(getComment().orElse(null))
+                .setExtraInfo(getExtraInfo().orElse(null))
+                .setHidden(isHidden())
+                .setProperties(getProperties());
+
+        return customizer.apply(builder).build();
```

Comment:
Rather than accept an argument for the `customizer`, I feel like it would be simpler to just call this method `toBuilder` or something and let the caller deal with setting the appropriate fields and calling `build()` themselves. Is there a specific reason you chose this approach?

Reply:
Nothing specific for doing it this way, I think it was based on earlier comment. But I agree, keeping it just builder is simpler. I have made the changes.